pkg/drivers/http/client.go

```diff
@@ -33,15 +34,52 @@ type client struct {
 
 const AuthBasic AuthMode = "Basic"
 
-func NewClient(isSecure bool, authMode AuthMode, cert string) Client {
+func NewClient(isSecure bool, authMode AuthMode, cert string, verifyHostName bool) Client {
 	glog.V(3).Info("NewClient():")
 	httpClient := &http.Client{Transport: http.DefaultTransport}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM([]byte(cert))
 	if isSecure {
-		if cert == "" {
-			httpClient.Transport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		/* #nosec */
+		if !verifyHostName {
+			httpClient.Transport.(*http.Transport).TLSClientConfig = &tls.Config{
+				RootCAs: caCertPool,
+				// Not actually skipping, we check the cert in VerifyPeerCertificate
+				InsecureSkipVerify: true,
+				VerifyPeerCertificate: func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+					// Code copy/pasted and adapted from
+					// https://github.com/golang/go/blob/81555cb4f3521b53f9de4ce15f64b77cc9df61b9/src/crypto/tls/handshake_client.go#L327-L344, but adapted to skip the hostname verification.
+					// See https://github.com/golang/go/issues/21971#issuecomment-412836078.
+
+					// If this is the first handshake on a connection, process and
+					// (optionally) verify the server's certificates.
+					certs := make([]*x509.Certificate, len(rawCerts))
+					for i, asn1Data := range rawCerts {
+						cert, err := x509.ParseCertificate(asn1Data)
+						if err != nil {
+							return err
+						}
+						certs[i] = cert
+					}
+
+					opts := x509.VerifyOptions{
+						Roots:         caCertPool,
+						CurrentTime:   time.Now(),
+						DNSName:       "", // <- skip hostname verification
+						Intermediates: x509.NewCertPool(),
+					}
+
+					for i, cert := range certs {
```

Comment:
nit, just for cleaner code but on the expense of creating a new slice for the duration of the loop:

```
for _, cert := range certs[1:] {
    ...
}
```

Reply:
if its copy paste code then nvm